docs/behat/basic-usage.rst

```diff
@@ -0,0 +1,540 @@
+            // Great benefit of using page objects is that we hide html manipulation behind a interfaces so we can inject different CreatePage which implements CreatePageInterface
+            // And have different html elements which allows for example chooseName($countryName).
+        }
+
+        /**
+         * @When I add it
+         */
+        public function iAddIt()
+        {
+            $this->createPage->create();
+        }
+
+        /**
+         * @Then /^the (country "([^"]+)") should appear in the store$/
+         */
+        public function countryShouldAppearInTheStore(CountryInterface $country) // This step use Country transformer to get Country object.
+        {
+            $this->indexPage->open();
+
+            //Webmozart assert library.
+            Assert::true(
+                $this->indexPage->isSingleResourceOnPage(['code' => $country->getCode()]),
+                sprintf('Country %s should exist but it does not', $country->getCode())
+            );
+        }
+    }
+
+.. code-block:: php
+
+    <?php
+
+    namespace Sylius\Behat\Context\Ui\Admin
+
+    final class NotificationContext implements Context
+    {
+        /**
+         * @var NotificationCheckerInterface
+         */
+        private $notificationChecker;
+        // This is a helper service which give access to proper notification elements.
+
+        /**
+         * @param NotificationCheckerInterface $notificationChecker
+         */
+        public function __construct(NotificationCheckerInterface $notificationChecker)
+        {
+            $this->notificationChecker = $notificationChecker;
+        }
+
+        /**
+         * @Then I should be notified that it has been successfully created
+         */
+        public function iShouldBeNotifiedItHasBeenSuccessfullyCreated()
+        {
+            $this->notificationChecker->checkNotification('has been successfully created.', NotificationType::success());
+        }
+    }
+
+Transformer contexts
+~~~~~~~~~~~~~~~~~~~~
+
+.. code-block:: php
+
+    <?php
+
+    namespace Sylius\Behat\Context\Transform;
+
+    final class CountryContext implements Context
+    {
+        /**
+         * @var CountryNameConverterInterface
+         */
+        private $countryNameConverter;
+
+        /**
+         * @var RepositoryInterface
+         */
+        private $countryRepository;
+
+        /**
+         * @param CountryNameConverterInterface $countryNameConverter
+         * @param RepositoryInterface $countryRepository
+         */
+        public function __construct(
+            CountryNameConverterInterface $countryNameConverter,
+            RepositoryInterface $countryRepository
+        ) {
+            $this->countryNameConverter = $countryNameConverter;
+            $this->countryRepository = $countryRepository;
+        }
+
+        /**
+         * @Transform /^country "([^"]+)"$/
+         * @Transform :country
+         * @Transform /^"([^"]+)" country$/
```

Comment:
Maybe it would be worth, to present two examples and discuss differences between them?